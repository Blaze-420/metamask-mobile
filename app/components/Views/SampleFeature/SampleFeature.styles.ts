import { StyleSheet } from 'react-native';
import { Colors } from '../../../util/theme/models';

const createStyles = (colors: Colors) =>
  StyleSheet.create({
    wrapper: {
      backgroundColor: colors.background.default,
      flex: 1,
      padding: 24,
      paddingBottom: 48,
    },
    heading: {
      marginTop: 16,
    },
    desc: {
      marginTop: 8,
    },
  });

export default createStyles;
