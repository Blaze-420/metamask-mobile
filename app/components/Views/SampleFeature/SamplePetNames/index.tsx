import React, {useState} from 'react';
import {KeyboardAvoidingView, Platform, SafeAreaView, View} from 'react-native';
import Text, {TextVariant} from '../../../../component-library/components/Texts/Text';
import {useStyles} from '../../../../component-library/hooks';
import styleSheet from './index.styles';
import {strings} from '../../../../../locales/i18n';
import {useSelector} from 'react-redux';
import {selectAddressBookByChain} from '../../../../selectors/addressBookController';
import {RootState} from '../../../../reducers';
import {PetNamesList} from './PetNamesList';
import {PetNamesForm} from './PetNamesForm';
import {Hex} from '@metamask/utils';
import {SupportedCaipChainId} from '@metamask/multichain-network-controller';

interface SamplePetNamesFormProps {
    chainId: SupportedCaipChainId | Hex;
}

export function SamplePetNames({chainId}: SamplePetNamesFormProps) {
    const {styles} = useStyles(styleSheet, {});

    const [selectedAddress, setSelectedAddress] = useState('');
    const [selectedName, setSelectedName] = useState('');

    const addressBook = useSelector((state: RootState) =>
        selectAddressBookByChain(
            state,
            chainId as Hex,
        ),
    );

    const onAccountPress = (params: { address: string; name: string }) => {
        setSelectedAddress(params.address);
        setSelectedName(params.name);
    };

    return (
        <SafeAreaView style={styles.wrapper}>
            <KeyboardAvoidingView
                style={styles.keyboardAvoidingView}
                behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
            >
                <View style={styles.card}>
                    <Text variant={TextVariant.HeadingSM}>{strings('sample_feature.pet_name.list_count_text')}</Text>
                    <PetNamesList addressBook={addressBook} onAccountPress={onAccountPress}/>
                    <PetNamesForm chainId={chainId} initialAddress={selectedAddress} initialName={selectedName}/>
                </View>
            </KeyboardAvoidingView>
        </SafeAreaView>
    );
}

