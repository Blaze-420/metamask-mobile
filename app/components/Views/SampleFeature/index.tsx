import React from 'react';
import {ScrollView} from 'react-native';

import {strings} from '../../../../locales/i18n';
import {useTheme} from '../../../util/theme';
import createStyles from './SampleFeature.styles';
import Text, {TextColor, TextVariant} from '../../../component-library/components/Texts/Text';
import {SampleCounterPane} from './SampleCounterPane/SampleCounterPane';
import {SampleNetworkDisplay} from './SampleNetworkDisplay/SampleNetworkDisplay';
import {useSelector} from 'react-redux';
import {selectNetworkImageSource, selectNetworkName} from '../../../selectors/networkInfos';
import {selectChainId, selectNetworkConfigurations} from '../../../selectors/networkController';
import {SamplePetnamesForm} from './SamplePetnamesForm/SamplePetnamesForm';


/**
 * Main view for app Experimental Settings
 */
const SampleFeature = () => {

    const theme = useTheme();
    const {colors} = theme;
    const styles = createStyles(colors);

    const networkImageSource = useSelector(selectNetworkImageSource);
    const chainId = useSelector(selectChainId);
    const networkConfigurations = useSelector(selectNetworkConfigurations);
    const name = useSelector(selectNetworkName);
    const networkName = networkConfigurations?.[chainId]?.name ?? name;

    return (
        <ScrollView style={styles.wrapper}>
            <Text
                color={TextColor.Default}
                variant={TextVariant.HeadingLG}
                style={styles.heading}
            >
                {strings('sample_feature.title')}
            </Text>
            <Text
                color={TextColor.Alternative}
                variant={TextVariant.BodyMD}
                style={styles.desc}
            >
                {strings('sample_feature.description')}
            </Text>
            <SampleNetworkDisplay
                name={networkName}
                imageSource={networkImageSource}
            />
            <SampleCounterPane/>
            <SamplePetnamesForm chainId={chainId}/>
        </ScrollView>
    );
};

export default SampleFeature;
