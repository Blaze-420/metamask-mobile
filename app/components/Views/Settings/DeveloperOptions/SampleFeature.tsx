import React from 'react';
import { useTheme } from '../../../../util/theme';
import Text, {
  TextVariant,
  TextColor,
} from '../../../../component-library/components/Texts/Text';
import { useStyles } from '../../../../component-library/hooks';
import styleSheet from './DeveloperOptions.styles';
import {strings} from '../../../../../locales/i18n';
import Button, {
    ButtonSize,
    ButtonVariants,
    ButtonWidthTypes
} from '../../../../component-library/components/Buttons/Button';
import {MetaMetricsEvents} from '../../../../core/Analytics';
import {useMetrics} from '../../../hooks/useMetrics';
import {useNavigation} from '@react-navigation/native';

function NavigateToSampleFeature() {
    const theme = useTheme();
    const { styles } = useStyles(styleSheet, { theme });
    const { trackEvent, createEventBuilder } = useMetrics();
    const navigation = useNavigation();

    const onPressNavigate = () => {
        trackEvent(
            createEventBuilder(MetaMetricsEvents.SETTINGS_SAMPLE_FEATURE).build(),
        );
        navigation.navigate('SampleFeature');
    };

    return (
        <>
            <Text
                color={TextColor.Alternative}
                variant={TextVariant.BodyMD}
                style={styles.desc}
            >
                {strings('app_settings.developer_options.sample_feature_desc')}
            </Text>
            <Button
                variant={ButtonVariants.Secondary}
                size={ButtonSize.Lg}
                label={strings('app_settings.developer_options.navigate_to_sample_feature')}
                onPress={onPressNavigate}
                width={ButtonWidthTypes.Full}
                style={styles.accessory}
            />
        </>
    );
}

export default function SampleFeature() {
  const theme = useTheme();
  const { styles } = useStyles(styleSheet, { theme });

  return (
    <>
        <Text
        color={TextColor.Default}
        variant={TextVariant.HeadingLG}
        style={styles.heading}
        >
            {strings('sample_feature.title')}
        </Text>
        <NavigateToSampleFeature />
    </>
  );
}
