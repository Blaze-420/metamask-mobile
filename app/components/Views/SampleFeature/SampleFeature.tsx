import React from 'react';
import {ScrollView} from 'react-native';

import {strings} from '../../../../locales/i18n';
import {useTheme} from '../../../util/theme';
import createStyles from './SampleFeature.styles';
import Text, {TextColor, TextVariant} from '../../../component-library/components/Texts/Text';
import {SampleCounterPane} from './SampleCounterPane/SampleCounterPane';
import {SampleNetworkDisplay} from './SampleNetworkDisplay/SampleNetworkDisplay';
import {SamplePetNames} from './SamplePetNames/SamplePetNames';
import useSampleNetwork from '../../hooks/SampleFeature/useSampleNetwork/useSampleNetwork';

/**
 * Main view for app Sample Feature
 *
 * @sampleFeature do not use in production code
 */
const SampleFeature = () => {

    const theme = useTheme();
    const {colors} = theme;
    const styles = createStyles(colors);

    const {networkName, networkImageSource} = useSampleNetwork();

    return (
        <ScrollView style={styles.wrapper}>
            <Text
                color={TextColor.Default}
                variant={TextVariant.HeadingLG}
                style={styles.heading}
            >
                {strings('sample_feature.title')}
            </Text>
            <Text
                color={TextColor.Alternative}
                variant={TextVariant.BodyMD}
                style={styles.desc}
            >
                {strings('sample_feature.description')}
            </Text>
            <SampleNetworkDisplay
                name={networkName}
                imageSource={networkImageSource}
            />
            <SampleCounterPane/>
            <SamplePetNames/>
        </ScrollView>
    );
};

export default SampleFeature;
