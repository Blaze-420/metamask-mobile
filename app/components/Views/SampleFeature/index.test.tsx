import React from 'react';
import renderWithProvider from '../../../util/test/renderWithProvider';
import SampleFeature from './';

jest.mock('react-native/Libraries/Linking/Linking', () => ({
    addEventListener: jest.fn(() => ({
        remove: jest.fn(),
    })),
}));

describe('SampleFeature', () => {
    it('render matches snapshot', () => {
        const {toJSON} = renderWithProvider(<SampleFeature/>);
        expect(toJSON()).toMatchSnapshot();
    });
});
