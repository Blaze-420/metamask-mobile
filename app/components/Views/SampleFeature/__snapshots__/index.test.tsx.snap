// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleFeature render matches snapshot 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "padding": 24,
      "paddingBottom": 48,
    }
  }
>
  <View>
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#141618",
          "fontFamily": "EuclidCircularB-Bold",
          "fontSize": 24,
          "fontWeight": "700",
          "letterSpacing": 0,
          "lineHeight": 32,
          "marginTop": 16,
        }
      }
    >
      Sample Feature
    </Text>
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#6a737d",
          "fontFamily": "EuclidCircularB-Regular",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 22,
          "marginTop": 8,
        }
      }
    >
      This is a sample feature as a template for developers.
    </Text>
  </View>
</RCTScrollView>
`;
